core/eval_test.go

```diff
@@ -61,82 +118,337 @@ func testEvalMSET(t *testing.T) {
+		"nil value": {
+			setup:  func() {},
+			input:  nil,
+			output: []byte(":0\r\n"),
+		},
+		"empty array": {
+			setup:  func() {},
+			input:  []string{},
+			output: []byte(":0\r\n"),
+		},
+		"key does not exist": {
+			setup:  func() {},
+			input:  []string{"NONEXISTENT_KEY"},
+			output: []byte(":0\r\n"),
+		},
+		"key exists": {
+			setup: func() {
+				key := "EXISTING_KEY"
+				value := "mock_value"
+				obj := &Obj{
+					Value:          value,
+					LastAccessedAt: uint32(time.Now().Unix()),
+				}
+				store[unsafe.Pointer(obj)] = obj
+				keypool[key] = unsafe.Pointer(obj)
+				KeyspaceStat[0]["keys"]++
+			},
+			input:  []string{"EXISTING_KEY"},
+			output: []byte(":1\r\n"),
+		},
+	}
+
+	runTests(t, tests, evalDEL)
 }
 
 // TestEvalPersist tests the evalPersist function using table-driven tests.
 func TestEvalPersist(t *testing.T) {
 	// Define test cases
-	tests := []struct {
-		name     string
-		args     []string
-		setup    func()
-		expected []byte
-	}{
-		{
-			name:     "wrong number of arguments",
-			args:     []string{"key1", "key2"},
-			expected: Encode(errors.New("ERR wrong number of arguments for 'persist' command"), false),
-		},
-		{
-			name:     "key does not exist",
-			args:     []string{"nonexistent"},
-			expected: RESP_ZERO,
-		},
-		{
-			name: "key exists but no expiration set",
-			args: []string{"existent_no_expiry"},
+	tests := map[string]testCase{
+		"wrong number of arguments": {
+			input:  []string{"key1", "key2"},
+			output: Encode(errors.New("ERR wrong number of arguments for 'persist' command"), false),
+		},
+		"key does not exist": {
+			input:  []string{"nonexistent"},
+			output: RESP_ZERO,
+		},
+		"key exists but no expiration set": {
+			input: []string{"existent_no_expiry"},
 			setup: func() {
 				evalSET([]string{"existent_no_expiry", "value"})
 			},
-			expected: RESP_MINUS_1,
+			output: RESP_MINUS_1,
 		},
-		{
-			name: "key exists and expiration removed",
-			args: []string{"existent_with_expiry"},
+		"key exists and expiration removed": {
+			input: []string{"existent_with_expiry"},
 			setup: func() {
 				evalSET([]string{"existent_with_expiry", "value", "EX", "1"})
 			},
-			expected: RESP_ONE,
+			output: RESP_ONE,
 		},
-		{
-			name: "key exists with expiration set and not expired",
-			args: []string{"existent_with_expiry_not_expired"},
+		"key exists with expiration set and not expired": {
+			input: []string{"existent_with_expiry_not_expired"},
 			setup: func() {
 				// Simulate setting a key with an expiration time that has not yet passed
 				evalSET([]string{"existent_with_expiry_not_expired", "value", "EX", "10000"}) // 10000 seconds in the future
 			},
-			expected: RESP_ONE,
+			output: RESP_ONE,
 		},
 	}
 
-	// Run test cases
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			// Setup test case
+	runTests(t, tests, evalPersist)
+}
+
+func runTests(t *testing.T, tests map[string]testCase, evalFunc func([]string) []byte) {
```

Comment:
Let's not use generic names for functions since they can pollute the package namespace. Maybe rename this to `runEvaltests`?